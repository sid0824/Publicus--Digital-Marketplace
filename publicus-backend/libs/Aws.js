import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});