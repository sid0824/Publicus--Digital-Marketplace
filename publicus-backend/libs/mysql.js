import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};