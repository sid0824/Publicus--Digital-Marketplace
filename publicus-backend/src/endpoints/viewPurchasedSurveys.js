import {runQuery} from "../../libs/mysql";
import {failure, json_success} from "../util/ResponseUtil";

//Generate SQL Query (Returns surveys that the user has purchased)
const selectPurchasedSurveyQuery = (cognitoId) => {
  return `SELECT s.*, (SELECT COUNT(*) FROM Results p WHERE p.surveyId = s.surveyId AND p.creatorId = '${cognitoId}' GROUP BY p.surveyId) as 'numResponses', `
    + `(SELECT AVG(r.rating) FROM Ratings r WHERE r.surveyId = s.surveyId AND r.userId = '${cognitoId}' GROUP BY r.surveyId) as 'myRating' `
    + `FROM Surveys s WHERE s.surveyId In `
    + "(SELECT surveyId FROM Licenses WHERE userId = '" + cognitoId + "' )";
};

export async function main(event, context) {
  try {
    const results = await runQuery(selectPurchasedSurveyQuery(event.requestContext.identity.cognitoIdentityId));
    return json_success(results);
  } catch (err) {
   console.error(err);
   return failure();
  }
}
