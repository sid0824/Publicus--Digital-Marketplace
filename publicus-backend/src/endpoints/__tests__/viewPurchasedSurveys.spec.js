import { main } from '../viewPurchasedSurveys';
import { runQuery } from "../../../libs/mysql";

jest.mock("../../../libs/mysql");

const mockEvent = {
  requestContext: {
    identity: {
      cognitoIdentityId: 'MOCK_COGNITO_ID'
    },
  },
  body: JSON.stringify({
  }),
  pathParameters: {
  }
};

describe('viewPurchasedSurveys', () => {

  beforeEach(() => {
  });

  afterEach(() => {
    runQuery.mockReset();
  });

  test(`should respond with failure when an error occurs`, async () => {
    runQuery.mockImplementation(() => {
      throw new Error('ERROR')
    });
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  });

  test(`should return all purchased surveys`, async () => {
    runQuery.mockResolvedValue([{ MOCK: 'FORM' }]);
    const response = await main(mockEvent);
    expect(runQuery.mock.calls[0][0]).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  })

});