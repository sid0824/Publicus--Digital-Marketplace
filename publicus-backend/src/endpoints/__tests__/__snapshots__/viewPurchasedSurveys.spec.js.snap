// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewPurchasedSurveys should respond with failure when an error occurs 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;

exports[`viewPurchasedSurveys should return all purchased surveys 1`] = `"SELECT s.*, (SELECT COUNT(*) FROM Results p WHERE p.surveyId = s.surveyId AND p.creatorId = 'MOCK_COGNITO_ID' GROUP BY p.surveyId) as 'numResponses', (SELECT AVG(r.rating) FROM Ratings r WHERE r.surveyId = s.surveyId AND r.userId = 'MOCK_COGNITO_ID' GROUP BY r.surveyId) as 'myRating' FROM Surveys s WHERE s.surveyId In (SELECT surveyId FROM Licenses WHERE userId = 'MOCK_COGNITO_ID' )"`;

exports[`viewPurchasedSurveys should return all purchased surveys 2`] = `
Object {
  "body": "[{\\"MOCK\\":\\"FORM\\"}]",
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "applications/json",
  },
  "statusCode": 200,
}
`;
